src/passes/Asyncify.cpp

```diff
@@ -455,8 +452,45 @@ class ModuleAnalyzer {
 
     map.swap(scanner.map);
 
+    std::set<Name> blacklist_names;
+    std::set<Name> whitelist_names;
+    std::set<std::string> blacklist_patterns;
+    std::set<std::string> whitelist_patterns;
+    std::set<std::string> blacklist_patterns_matched;
+    std::set<std::string> whitelist_patterns_matched;
+
+    // The lists contain human-readable strings. Turn them into the internal
+    // escaped names for later comparisons
+    auto processList = [&module](const String::Split& list,
```

Comment:
I think this is a good idea, but it would be clearer as a standalone class. What I mean is something like `class PatternMatcher` whose constructor takes a list of strings (that may have `*` in some of them), does all the work of `processList` in the constructor, and then provides a `match` function to test a value.

Reply:
It seems this will make the reporting harder.
I believe I kept the level of complexity/clarity on-par with the initial code, so that sounds like a code clean-up/refactoring that is distinct and can be done later, what do you think?

Reply:
It could be done separately, I agree.

But I do worry about increasing the code complexity in this PR without a timeline of when we can do this refactoring to get things in good shape again. When do you think you'd have time for it?

Reply:
Done.